graphql-dgs-subscriptions-sse/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/sse/DgsSSESubscriptionHandler.kt

```diff
@@ -44,119 +49,73 @@ import java.util.*
 @RestController
 open class DgsSSESubscriptionHandler(open val dgsQueryExecutor: DgsQueryExecutor) {
 
-    @RequestMapping("/subscriptions", produces = ["text/event-stream"])
-    fun subscriptionWithId(@RequestParam("query") queryBase64: String): ResponseEntity<SseEmitter> {
-        val emitter = SseEmitter(-1)
-        val sessionId = UUID.randomUUID().toString()
+    @RequestMapping("/subscriptions", produces = [MediaType.TEXT_EVENT_STREAM_VALUE])
+    fun subscriptionWithId(@RequestParam("query") queryBase64: String): Flux<ServerSentEvent<String>> {
         val query = try {
             String(Base64.getDecoder().decode(queryBase64), StandardCharsets.UTF_8)
         } catch (ex: IllegalArgumentException) {
-            emitter.send("Error decoding base64 encoded query")
-            emitter.complete()
-            return ResponseEntity.badRequest().body(emitter)
+            throw ServerWebInputException("Error decoding base64-encoded query")
         }
 
         val queryPayload = try {
             mapper.readValue(query, QueryPayload::class.java)
         } catch (ex: Exception) {
-            emitter.send("Error parsing query: ${ex.message}")
-            emitter.complete()
-            return ResponseEntity.badRequest().body(emitter)
+            throw ServerWebInputException("Error parsing query: ${ex.message}")
+        }
+
+        if (!isSubscriptionQuery(queryPayload.query)) {
+            throw ServerWebInputException("Invalid query. operation type is not a subscription")
         }
 
         val executionResult: ExecutionResult = dgsQueryExecutor.execute(queryPayload.query, queryPayload.variables)
         if (executionResult.errors.isNotEmpty()) {
-            return if (
-                executionResult.errors.asSequence().filterIsInstance<ValidationError>().any() ||
-                executionResult.errors.asSequence().filterIsInstance<InvalidSyntaxError>().any()
-            ) {
-                val errorMessage = "Subscription query failed to validate: ${executionResult.errors.joinToString(", ")}"
-                emitter.send(errorMessage)
-                emitter.complete()
-                ResponseEntity.badRequest().body(emitter)
+            val errorMessage = if (executionResult.errors.any { error -> error is ValidationError || error is InvalidSyntaxError }) {
+                "Subscription query failed to validate: ${executionResult.errors.joinToString()}"
             } else {
-                val errorMessage = "Error executing subscription query: ${executionResult.errors.joinToString(", ")}"
-                logger.error(errorMessage)
-                emitter.send(errorMessage)
-                emitter.complete()
-                ResponseEntity.status(500).body(emitter)
+                "Error executing subscription query: ${executionResult.errors.joinToString()}"
             }
-        }
-
-        val subscriber = object : Subscriber<ExecutionResult> {
-            lateinit var subscription: Subscription
```

Comment:
I think this actually wasn't thread safe, since the property wasn't volatile / access wasn't synchronized, although in practice I never saw any actual issues.